includes/class-newspack-newsletters-contacts.php

```diff
@@ -139,7 +141,24 @@ public static function delete( $user_id ) {
 		if ( ! method_exists( $provider, 'delete_contact' ) ) {
 			return new WP_Error( 'newspack_newsletters_invalid_provider_method', __( 'Provider does not support deleting user subscriptions.' ) );
 		}
-		return $provider->delete_contact( $user->user_email );
+		$result = $provider->delete_contact( $user->user_email );
+
+		do_action(
+			'newspack_log',
+			'newspack_esp_sync_delete_contact',
+			$context,
+			[
+				'type'       => is_wp_error( $result ) ? 'error' : 'debug',
+				'data'       => [
+					'provider' => $provider->service,
+					'errors'   => is_wp_error( $result ) ? $result->get_error_message() : [],
+				],
+				'user_email' => $email,
```

Comment:
```suggestion
				'user_email' => $user->user_email,
```

Getting a PHP warning: `PHP Warning:  Undefined variable $email in /srv/htdocs/wp-content/plugins/newspack-newsletters/includes/class-newspack-newsletters-contacts.php on line 156`